test/sys/test_sockopt.rs

```diff
@@ -196,3 +196,41 @@ fn test_ttl_opts() {
     setsockopt(fd6, sockopt::Ipv6Ttl, &1)
         .expect("setting ipv6ttl on an inet6 socket should succeed");
 }
+
+#[test]
+#[cfg(any(target_os = "ios", target_os = "macos"))]
+fn test_dontfrag_opts() {
+    let fd4 = socket(AddressFamily::Inet, SockType::Stream, SockFlag::empty(), SockProtocol::Tcp).unwrap();
+    setsockopt(fd4, sockopt::IpDontFrag, &true)
+        .expect("setting IP_DONTFRAG on an inet stream socket should succeed");
+    setsockopt(fd4, sockopt::IpDontFrag, &false)
+        .expect("unsetting IP_DONTFRAG on an inet stream socket should succeed");
+    let fd4d = socket(AddressFamily::Inet, SockType::Datagram, SockFlag::empty(), None).unwrap();
+    setsockopt(fd4d, sockopt::IpDontFrag, &true)
+        .expect("setting IP_DONTFRAG on an inet datagram socket should succeed");
+    setsockopt(fd4d, sockopt::IpDontFrag, &false)
+        .expect("unsetting IP_DONTFRAG on an inet datagram socket should succeed");
+}
+
+#[test]
+#[cfg(all(
+    any(target_arch = "x86", target_arch = "x86_64"),
+    any(
+        target_os = "android",
+        target_os = "ios",
+        target_os = "linux",
+        target_os = "macos",
+    )
+))]
```

Comment:
This logic is incorrect - why is the target architecture limited?

There aren't any x86 iOS targets, and the test passes when I run it locally on Android via Termux.

Reply:
This is because some CI test failed. Let me take a look again.

Reply:
Updated to exclude this test on qemu-based CI.